File: leet_953_verify_dict.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        E = { c : i for i, c in enumerate(order) }
        size = len(words) - 1
        for i in range(size):
            L, R = words[i], words[i + 1]
            Lz = len(L)
            Rz = len(R)
            for j in range(len(L)):
                if j == Rz:
                    return False
                if L[j] != R[j]:
                    if E[ L[j] ] > E[ R[j] ]:
                        return False
                    break
        return True 

File: leet_953_verify_dict.js
/**
 * @param {string[]} words
 * @param {string} order
 * @return {boolean}
 */

var isAlienSorted = function(words, order) {
    let dict = new Object()
    let oz = order.length
    let wz = words.length
    let i = -1

    while (++i < oz) dict[ order[i] ] = i
    i = -1
    while (++i < wz - 1) {
        let L = words[i], R = words[i + 1]
        let Ls = L.length, Rs = R.length
        let j = -1
        while (++j < Ls) {
            if (Rs == j)    return false
            if (L[j] != R[j]) {
                if (dict[L[j]] > dict[R[j]] )   return false
                break
            }
        }
    }
    return true 
};
